packages/analytics/ios/RNFBAnalytics/RNFBAnalyticsModule.m

```diff
@@ -135,4 +135,23 @@ - (dispatch_queue_t)methodQueue {
     return resolve([NSNull null]);
   }
 
+#pragma mark -
+#pragma mark Private methods
+
+  - (NSDictionary *)cleanJavascriptParams:(NSDictionary *)params {
+    NSMutableDictionary *newParams = [params mutableCopy];
+    if (newParams[kFIRParameterItems]) {
+      NSMutableArray *newItems = [NSMutableArray array];
+      [(NSArray *)newParams[kFIRParameterItems] enumerateObjectsUsingBlock:^(id  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+        NSMutableDictionary *item = [obj mutableCopy];
+        if (item[kFIRParameterQuantity]) {
+          item[kFIRParameterQuantity] = @([item[kFIRParameterQuantity] integerValue]);
+        }
+        [newItems addObject:[item copy]];
+      }];
+      newParams[kFIRParameterItems] = [newItems copy];
+    }
+    return [newParams copy];
+  }
```

Comment:
Alternative solution would be to transform the incoming dictionary into a JSON string, search and replace all numbers with zeroes after the dot with whole numbers and transform it back into an object, but obviously this may have some unexpected impact on other things...